src/platform/Darwin/PlatformManagerImpl.h

```diff
@@ -75,13 +77,38 @@ class PlatformManagerImpl final : public PlatformManager,
     CHIP_ERROR _StopEventLoopTask();
 
     void _RunEventLoop();
-    void _LockChipStack(){};
-    bool _TryLockChipStack() { return false; };
-    void _UnlockChipStack(){};
+    void _LockChipStack()
+    {
+        mLock.lock();
```

Comment:
@bzbarsky-apple I am not sure about darwin internals ... I thought darwin explicitly has no stack lock. Could you take a look at these changes?